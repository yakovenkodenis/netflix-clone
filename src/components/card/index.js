import React, { useState, useContext } from 'react';

import { CardFeatureContext } from '../../context/cardFeature';

import {
    Container,
    Content,
    Title,
    Subtitle,
    Text,
    Meta,
    Item,
    Group,
    Image,
    Entities,
    Feature,
    FeatureTitle,
    FeatureText,
    FeatureClose,
    Maturity,
} from './styles/card';

export default function Card({ children, ...restProps }) {
    const [showFeature, setShowFeature] = useState(false);
    const [itemFeature, setItemFeature] = useState({});

    return (
        <CardFeatureContext.Provider
            value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
        >
            <Container {...restProps}>{children}</Container>
        </CardFeatureContext.Provider>
    );
}

Card.Group = function CardGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>;
};

Card.Title = function CardTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>;
};

Card.Subtitle = function CardSubtitle({ children, ...restProps }) {
    return <Subtitle {...restProps}>{children}</Subtitle>;
};

Card.Text = function CardText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>;
};

Card.Meta = function CardMeta({ children, ...restProps }) {
    return <Meta {...restProps}>{children}</Meta>;
};

Card.Entities = function CardEntities({ children, ...restProps }) {
    return <Entities {...restProps}>{children}</Entities>;
};

Card.Feature = function CardFeature({ category, children, ...restProps }) {
    const { showFeature, itemFeature, setShowFeature } = useContext(
        CardFeatureContext
    );
    return showFeature ? (
        <Feature
            {...restProps}
            src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
        >
            <Content>
                <FeatureTitle>{itemFeature.title}</FeatureTitle>
                <FeatureText>{itemFeature.description}</FeatureText>
                <FeatureClose onClick={() => setShowFeature(false)}>
                    <img src="/images/icons/close.png" alt="Close" />
                </FeatureClose>

                <Group margin="30px 0" flexDirection="row" alignItems="center">
                    <Maturity rating={itemFeature.maturity}>
                        {itemFeature.maturity < 12
                            ? 'PG'
                            : itemFeature.maturity}
                    </Maturity>
                    <FeatureText fontWeight="bold">
                        {itemFeature.genre.charAt(0).toUpperCase() +
                            itemFeature.genre.slice(1)}
                    </FeatureText>
                </Group>
                {children}
            </Content>
        </Feature>
    ) : null;
};

Card.Item = function CardItem({ item, children, ...restProps }) {
    const { setShowFeature, setItemFeature } = useContext(CardFeatureContext);

    return (
        <Item
            onClick={() => {
                setItemFeature(item);
                setShowFeature(true);
            }}
            {...restProps}
        >
            {children}
        </Item>
    );
};

Card.Image = function CardImage({ ...restProps }) {
    return <Image {...restProps} />;
};
