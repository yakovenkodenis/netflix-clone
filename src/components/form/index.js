import React from 'react';

import {
    Container,
    Base,
    Error,
    Title,
    Text,
    TextSmall,
    Link,
    Input,
    Submit,
} from './styles/form';

export default function Form({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
    return <Error {...restProps}>{children}</Error>;
};

Form.Base = function BaseForm({ children, ...restProps }) {
    return <Base {...restProps}>{children}</Base>;
};

Form.Title = function TitleForm({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>;
};

Form.TextSmall = function TextSmallForm({ children, ...restProps }) {
    return <TextSmall {...restProps}>{children}</TextSmall>;
};

Form.Text = function TextForm({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>;
};

Form.Link = function LinkForm({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>;
};

Form.Input = function InputForm({ children, ...restProps }) {
    return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function SubmitForm({ children, ...restProps }) {
    return <Submit {...restProps}>{children}</Submit>;
};
