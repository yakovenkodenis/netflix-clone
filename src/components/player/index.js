import React, { useState, useContext } from 'react';
import { createPortal } from 'react-dom';

import { PlayerContext } from '../../context/player';

import { Container, Button, Overlay, Inner, Close } from './styles/player';

export default function Player({ children, ...restProps }) {
    const [showPlayer, setShowPlayer] = useState(false);
    return (
        <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
            <Container {...restProps}>{children}</Container>
        </PlayerContext.Provider>
    );
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
    const { showPlayer, setShowPlayer } = useContext(PlayerContext);
    return showPlayer
        ? createPortal(
              <Overlay onClick={() => setShowPlayer(false)} {...restProps}>
                  <Inner>
                      <video id="netflix-palyer" controls>
                          <source src={src} type="video/mp4" />
                      </video>
                      <Close />
                  </Inner>
              </Overlay>,
              document.body
          )
        : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
    const { setShowPlayer } = useContext(PlayerContext);
    return (
        <Button
            onClick={() => setShowPlayer((prevShowPlayer) => !prevShowPlayer)}
            {...restProps}
        >
            Play
        </Button>
    );
};
