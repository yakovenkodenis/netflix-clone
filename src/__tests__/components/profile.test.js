import React from 'react';
import { render } from '@testing-library/react';

import { Profiles } from '../../components';

describe('<Profiles />', () => {
    it('renders the <Profiles /> with populated data', () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Who's watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.Item onClick={() => {}}>
                        <Profiles.Picture
                            src={'/images/whatever.jpg'}
                            data-testid="profile-picture"
                        />
                        <Profiles.Name>Test Name</Profiles.Name>
                    </Profiles.Item>
                </Profiles.List>
                <Profiles.User data-testid="user-stub" />
            </Profiles>
        );

        expect(getByText("Who's watching?")).toBeTruthy();
        expect(getByText('Test Name')).toBeTruthy();
        expect(getByTestId('profile-picture')).toBeTruthy();
        expect(getByTestId('user-stub')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the <Profiles /> with populated data but misc profile image', () => {
        const { container, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Who's watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.Item onClick={() => {}}>
                        <Profiles.Picture
                            src={null}
                            data-testid="profile-picture-misc"
                        />
                        <Profiles.Name>Test Name</Profiles.Name>
                    </Profiles.Item>
                </Profiles.List>
            </Profiles>
        );

        expect(getByTestId('profile-picture-misc')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});
