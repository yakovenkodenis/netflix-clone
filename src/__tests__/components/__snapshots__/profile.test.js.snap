// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 glIXWo"
>
  <h1
    class="profiles__Title-bxtkkv-1 iIBFxC"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 eKtfuL"
  >
    <li
      class="profiles__Item-bxtkkv-6 gumqJc"
    >
      <img
        class="profiles__Picture-bxtkkv-4 izXQyg"
        data-testid="profile-picture"
        src="/images/users//images/whatever.jpg.png"
      />
      <p
        class="profiles__Name-bxtkkv-5 laQTyZ"
      >
        Test Name
      </p>
    </li>
  </ul>
  <div
    class="profiles__User-bxtkkv-3"
    data-testid="user-stub"
  />
</div>
`;

exports[`<Profiles /> renders the <Profiles /> with populated data but misc profile image 1`] = `
<div
  class="profiles__Container-bxtkkv-0 glIXWo"
>
  <h1
    class="profiles__Title-bxtkkv-1 iIBFxC"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 eKtfuL"
  >
    <li
      class="profiles__Item-bxtkkv-6 gumqJc"
    >
      <img
        class="profiles__Picture-bxtkkv-4 izXQyg"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-bxtkkv-5 laQTyZ"
      >
        Test Name
      </p>
    </li>
  </ul>
</div>
`;
